eng/scripts/Pack-Crates.ps1

```diff
@@ -0,0 +1,330 @@
+
+    foreach ($name in $toml.dependencies.Keys) {
+      # we want to look at the dependency as it was resolved by `cargo metadata`
+      # this will resolve workspace depdencies, but retain their path/no-version state 
+      $dependency = $package.dependencies | Where-Object -Property name -EQ -Value $name | Select-Object -First 1
+      # If the dependency is a path dependency, set the version to the version of the package in the workspace
+      if ($dependency.path -and !$dependency.version) {
+        $tomlDependency = $toml.dependencies.$name
+        $dependencyVersion = $packages | Where-Object -Property name -EQ -Value $name | Select-Object -ExpandProperty version -First 1
+
+        $tomlDependency.version = $dependencyVersion
+        $dirty = $true
+      }
+    }
+    if ($dirty) {
+      $toml | ConvertTo-Toml -Depth 10 | Set-Content -Path $Package.manifest_path -Encoding utf8
+    }
+  }
+}
+
+function Get-ApiMetadata($package) {
+  $packagePath = Split-Path -Path $package.manifest_path -Parent
+  $readmePath = Join-Path -Path $packagePath -ChildPath $package.readme
+
+  $jsonBody = [ordered]@{
+    'name'          = $package.name
+    'vers'          = $package.version
+    'deps'          = @()
+    'features'      = $package.features
+    'authors'       = $package.authors
+    'description'   = $package.description
+    'documentation' = $package.documentation
+    'homepage'      = $package.homepage
+    'readme'        = if (Test-Path -Path $readmePath) {
+      Get-Content -Path $readmePath -Raw
+    }
+    else {
+      $null
+    }
+    'readme_file'   = $package.readme
+    'keywords'      = $package.keywords
+    'categories'    = $package.categories
+    'license'       = $package.license
+    'license_file'  = $package.license_file
+    'repository'    = $package.repository
+    'links'         = $package.links
+    'rust_version'  = $package.rust_version
+  }
+
+  foreach ($dependency in $package.dependencies) {
+    $jsonBody.deps += @{
+      'name'                  = $dependency.name
+      'version_req'           = $dependency.req
+      'features'              = $dependency.features
+      'optional'              = $dependency.optional
+      'default_features'      = $dependency.default_features
+      'target'                = $dependency.target
+      'kind'                  = $dependency.kind
+      'explicit_name_in_toml' = $dependency.rename
+    }
+  }
+
+  return $jsonBody
+}
+
+function New-ApiPutFile($crateMetadata, $CrateFilePath) {
+  $metadataBytes = [Text.Encoding]::Utf8.GetBytes($crateMetadata)
+  $metadataLengthBytes = [BitConverter]::GetBytes([UInt32]$metadataBytes.Length)
+  $crateBytes = [IO.File]::ReadAllBytes($CrateFilePath)
+  $crateLengthBytes = [BitConverter]::GetBytes([UInt32]$crateBytes.Length)
+
+  $bytes += $metadataLengthBytes + $metadataBytes + $crateLengthBytes + $crateBytes
+
+  return $bytes
+}
+
+Push-Location $RepoRoot
```

Comment:
Just use `-p` on most `cargo` commands. There's no value in pushing into directories. Just more work for no gain.

Reply:
I do use `--package $packageName` on the cargo commands, I just do it from the repo root